### ISSUE TITLE
Exception when downloading from FH Dortmund ILIAS

### ISSUE THREAD

**Issue by tobire** [thetric/ilias-downloader-cli#19]
The following exception occurs, when I try to download content from the ILIAS of the FH Dortmund:

```
Exception in thread "main" com.github.thetric.iliasdownloader.connector.domparser.impl.IliasHttpException: Failed to GET https://www.ilias.fh-dortmund.de/ilias/ilias.php?baseClass=ilPersonalDesktopGUI&cmd=jumpToSelectedItems: 
	at com.github.thetric.iliasdownloader.connector.domparser.impl.webclient.OkHttpIliasWebClient.checkResponse(OkHttpIliasWebClient.kt:94)
	at com.github.thetric.iliasdownloader.connector.domparser.impl.webclient.OkHttpIliasWebClient.getHtml(OkHttpIliasWebClient.kt:81)
	at com.github.thetric.iliasdownloader.connector.domparser.impl.course.CourseSyncServiceImpl.getHtml(CourseSyncServiceImpl.kt:38)
	at com.github.thetric.iliasdownloader.connector.domparser.impl.course.CourseSyncServiceImpl.connectAndGetDocument(CourseSyncServiceImpl.kt:33)
	at com.github.thetric.iliasdownloader.connector.domparser.impl.course.CourseSyncServiceImpl.getJoinedCourses(CourseSyncServiceImpl.kt:28)
	at com.github.thetric.iliasdownloader.connector.domparser.impl.WebIliasService.getJoinedCourses(WebIliasService.kt:30)
	at com.github.thetric.iliasdownloader.cli.UserPreferencesUpdateServiceImpl.updateSyncCourses(UserPreferencesUpdateServiceImpl.kt:38)
	at com.github.thetric.iliasdownloader.cli.UserPreferencesUpdateServiceImpl.updatePreferences(UserPreferencesUpdateServiceImpl.kt:27)
	at com.github.thetric.iliasdownloader.cli.CliController.startCliController(CliController.kt:36)
	at com.github.thetric.iliasdownloader.cli.CliKt.main(Cli.kt:36)
```

**Comment by thetric** [comment]
Should be fixed in the latest release: https://github.com/thetric/ilias-downloader-cli/releases/tag/v4.0.2

**Comment by tobire** [comment]
Now there are no courses available to select. My ILIAS "homepage" with the course list is at https://www.ilias.fh-dortmund.de/ilias/ilias.php?baseClass=ilPersonalDesktopGUI&cmd=jumpToSelectedItems

**Comment by thetric** [comment]
I managed to fix the login but the CLI fails retrieve a file list due to some changes in the WebDAV interface:

```
Logging in at https://www.ilias.fh-dortmund.de/ilias/ilias.php?lang=en&client_id=ilias-fhdo&cmd=post&cmdClass=ilstartupgui&cmdNode=yc&baseClass=ilStartUpGUI&rtoken= 
Login at https://www.ilias.fh-dortmund.de/ilias/ilias.php?lang=en&client_id=ilias-fhdo&cmd=post&cmdClass=ilstartupgui&cmdNode=yc&baseClass=ilStartUpGUI&rtoken= succeeded  
Get all courses and groups from https://www.ilias.fh-dortmund.de/ilias/ilias.php?baseClass=ilPersonalDesktopGUI&cmd=jumpToSelectedItems 
Syncing 1 courses: Softwaretechnik 
Sync started 
Exception in thread "main" com.github.thetric.iliasdownloader.connector.domparser.IliasHttpException: Failed to GET https://www.ilias.fh-dortmund.de/ilias/webdav.php/ilias-fhdo/ref_86572/: HTTP status code 401
	at com.github.thetric.iliasdownloader.connector.domparser.IliasWebClient.checkResponse(IliasWebClient.kt:88)
	at com.github.thetric.iliasdownloader.connector.domparser.IliasWebClient.getHtml(IliasWebClient.kt:75)
	at com.github.thetric.iliasdownloader.connector.domparser.CourseSyncService.getHtml(CourseSyncService.kt:33)
	at com.github.thetric.iliasdownloader.connector.domparser.CourseSyncService.getItemContainersFromUrl(CourseSyncService.kt:58)
	at com.github.thetric.iliasdownloader.connector.domparser.CourseSyncService.visit(CourseSyncService.kt:46)
	at com.github.thetric.iliasdownloader.connector.domparser.WebIliasService.visit(WebIliasService.kt:35)
	at com.github.thetric.iliasdownloader.cli.SyncController.executeSync(SyncController.kt:41)
	at com.github.thetric.iliasdownloader.cli.SyncController.startSync(SyncController.kt:28)
	at com.github.thetric.iliasdownloader.cli.CliController.startCliController(CliController.kt:44)
	at com.github.thetric.iliasdownloader.cli.CliKt.main(Cli.kt:32)
```

In the browser the link www.ilias.fh-dortmund.de/ilias/webdav.php/ilias-fhdo/ref_86572/ keeps redirecting to the mounting instructions :(

**Comment by thetric** [comment]
I released [4.0.3](https://github.com/thetric/ilias-downloader-cli/releases/tag/v4.0.3) to fix the login issue but the courses still cannot be downloaded

**Comment by thetric** [comment]
Fixed in https://github.com/thetric/ilias-downloader-cli/releases/tag/v4.0.4
